web_src/js/markup/render-iframe.ts

```diff
@@ -0,0 +1,22 @@
+import {generateElemId, queryElemChildren} from '../utils/dom.ts';
+import {isDarkTheme} from '../utils.ts';
+
+export async function loadRenderIframeContent(iframe: HTMLIFrameElement) {
+  const iframeSrcUrl = iframe.getAttribute('data-src');
+  if (!iframe.id) iframe.id = generateElemId('gitea-iframe-');
+
+  window.addEventListener('message', (e) => {
+    if (e.data && e.data.giteaIframeCmd === 'resize' && e.data.giteaIframeId === iframe.id) {
+      iframe.style.height = `${e.data.giteaIframeHeight}px`;
+    }
+  });
```

Comment:
Wouldn't the same mechanism we use in the mermaid iframe work to update the height correctly?

https://github.com/go-gitea/gitea/blob/195fc715ff6a8c67c1b3473316b52379bfca6d49/web_src/js/markup/mermaid.ts#L60

That and the `IntersectionObserver`.

Reply:
Quite sad that the web platform apparently does not offer anything natively to size iframe to its content.